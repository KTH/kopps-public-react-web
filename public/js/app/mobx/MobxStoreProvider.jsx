/* eslint no-use-before-define: ["error", "nofunc"] */

import React from 'react'

import { createMobxManager } from './manager'

export default MobxStoreProvider

/**
 * React-component which prepares a MobX-store
 * so that its data can be used with the useStore() hook
 * of any other component embedded into this Provider.
 *
 * @param {object} props
 * @param {function} props.initCallback
 * @param {string} [props.storeId]
 * @param {object} props.children
 *
 */
function MobxStoreProvider(props) {
  const { initCallback, storeId, children } = props

  const manager = createMobxManager(storeId || 'default', initCallback)

  const { Provider: ReactContextProvider, value: mobxStore } = manager.getProviderComponentAndValue()

  return <ReactContextProvider value={mobxStore}>{children}</ReactContextProvider>
}
