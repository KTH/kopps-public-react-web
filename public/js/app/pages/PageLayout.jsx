import React from 'react'
import { Col, Row } from 'reactstrap'
import MainMenu from '../components/MainMenu'

function MainContent({ children }) {
  return <main id="mainContent">{children}</main>
}

function PageLayout({ children }) {
  return (
    <Row style={{ padding: '30px 0' }}>
      <MainMenu />
      <Col>
        <MainContent>{children}</MainContent>
      </Col>
    </Row>
  )
}

export default PageLayout
