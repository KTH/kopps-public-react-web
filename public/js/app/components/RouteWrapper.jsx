/* eslint-disable react/jsx-props-no-spreading */
import React from 'react'
import { Route } from 'react-router-dom'
import PropTypes from 'prop-types'
import { MobxStoreProvider } from '../mobx'

function RouteWrapper({
  createBreadcrumbs,
  component: Component,
  layout: Layout,
  applicationStore,
  createMenuData,
  ...rest
}) {
  const menuData = createMenuData(applicationStore)
  const breadcrumbs = createBreadcrumbs(applicationStore)
  return (
    <MobxStoreProvider initCallback={() => applicationStore}>
      <Route
        {...rest}
        render={() => (
          <Layout breadcrumbs={breadcrumbs} menuData={menuData}>
            <Component />
          </Layout>
        )}
      />
    </MobxStoreProvider>
  )
}

RouteWrapper.propTypes = {
  createBreadcrumbs: PropTypes.func,
  component: PropTypes.oneOfType([PropTypes.node, PropTypes.func, PropTypes.elementType]).isRequired,
  layout: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
  createMenuData: PropTypes.func.isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  applicationStore: PropTypes.object.isRequired,
}

RouteWrapper.defaultProps = {
  createBreadcrumbs: () => ({ include: 'student', items: [] }),
}

export default RouteWrapper
