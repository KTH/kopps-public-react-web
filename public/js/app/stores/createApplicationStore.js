/* eslint no-use-before-define: ["error", "nofunc"] */

// @ts-check

// eslint-disable-next-line no-unused-vars
import { observable } from 'mobx'
import commonStore from './commonStore'
import curriculumStore from './curriculumStore'
import studyProgrammeStore from './studyProgrammeStore'
import searchCoursesStore from './searchCoursesStore'
import appendix1Store from './appendix1Store'
import appendix2Store from './appendix2Store'
import literatureStore from './literatureStore'

export default createApplicationStore

const store = {
  /**
   * @property {string} message
   */
  message: 'Hallo',
  /**
   * @method
   * @param {string} text
   */
  setMessage,
  /**
   * @property {string} lastAdmissionTerm
   */
  lastAdmissionTerm: '',
  /**
   * @method
   * @param {string} lastAdmissionTerm
   */
  setLastAdmissionTerm,
  /**
   * @method
   * @param {map<string, {}>} programmes
   */
  setProgrammes,

  /**
   * @property {map<string, {}>} programmes
   */
  programmes: [],

  /**
   * @method
   * @param {[]} currentSchoolsWithDepartments
   */
  setCurrentSchoolsWithDepartments,

  /**
   * @property {[]} currentSchoolsWithDepartments
   */
  currentSchoolsWithDepartments: [],

  /**
   * @method
   * @param {[]} deprecatedSchoolsWithDepartments
   */
  setDeprecatedSchoolsWithDepartments,

  /**
   * @property {[]} deprecatedSchoolsWithDepartments
   */
  deprecatedSchoolsWithDepartments: [],
  /**
   * @property {string} departmentName
   */
  departmentName: '',
  /**
   * @method
   * @param {string} departmentName
   */
  setDepartmentName,

  /**
   * @property {[]} departmentCourses
   */
  departmentCourses: [],
  /**
   * @method
   * @param {[]} departmentCourses
   */
  setDepartmentCourses,
  /**
   * @property {string} programmeName
   */
  programmeName: '',
  /**
   * @method
   * @param {string} programmeName
   */
  setProgrammeName,
  /**
   * @property {string} programmeCode
   */
  programmeCode: '',
  /**
   * @method
   * @param {string} programmeCode
   */
  setProgrammeCode,
  /**
   * @property {[]} programmeTerms
   */
  programmeTerms: [],
  /**
   * @method
   * @param {[]} programmeTerms
   */
  setProgrammeTerms,
  /**
   * @property {number} lengthInStudyYears
   */
  lengthInStudyYears: 0,
  /**
   * @method
   * @param {number} lengthInStudyYears
   */
  setLengthInStudyYears,
  /**
   * @property {string} thisHostBaseUrl
   */
  thisHostBaseUrl: null,
}

function createApplicationStore(storeId) {
  switch (storeId) {
    case 'curriculum':
      return {
        ...commonStore,
        ...curriculumStore,
      }
    case 'searchCourses':
      return {
        ...commonStore,
        ...searchCoursesStore,
      }
    case 'objectives':
    case 'extent':
    case 'eligibility':
    case 'implementation':
      return {
        ...commonStore,
        ...studyProgrammeStore,
      }
    case 'appendix1':
      return {
        ...commonStore,
        ...studyProgrammeStore,
        ...appendix1Store,
      }
    case 'appendix2':
      return {
        ...commonStore,
        ...studyProgrammeStore,
        ...appendix2Store,
      }
    case 'literatureList':
      return {
        ...commonStore,
        ...literatureStore,
      }
    default:
      return {
        ...commonStore,
        ...store,
      }
  }
}

function setMessage(text = 'Happy coding!! :)') {
  this.message = text
}

function setProgrammes(programmes) {
  this.programmes = [...programmes]
}

function setCurrentSchoolsWithDepartments(currentSchoolsWithDepartments) {
  this.currentSchoolsWithDepartments = currentSchoolsWithDepartments
}

function setDeprecatedSchoolsWithDepartments(deprecatedSchoolsWithDepartments) {
  this.deprecatedSchoolsWithDepartments = deprecatedSchoolsWithDepartments
}

function setDepartmentName(departmentName) {
  this.departmentName = departmentName
}

function setDepartmentCourses(departmentCourses) {
  this.departmentCourses = departmentCourses
}

function setLastAdmissionTerm(lastAdmissionTerm) {
  this.lastAdmissionTerm = lastAdmissionTerm
}

function setProgrammeName(programmeName) {
  this.programmeName = programmeName
}

function setProgrammeCode(programmeCode) {
  this.programmeCode = programmeCode
}

function setProgrammeTerms(programmeTerms) {
  this.programmeTerms = programmeTerms
}

function setLengthInStudyYears(lengthInStudyYears) {
  this.lengthInStudyYears = lengthInStudyYears
}
