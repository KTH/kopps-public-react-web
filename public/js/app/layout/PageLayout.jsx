import React, { useEffect } from 'react'
import ReactDOM from 'react-dom'
import { Col, Row } from 'reactstrap'
import { Breadcrumbs } from '@kth/kth-reactstrap/dist/components/utbildningsinfo'

import { useStore } from '../mobx'
import MainMenu from '../components/MainMenu'

function renderBreadcrumbsIntoKthHeader(language) {
  const breadcrumbContainer = document.getElementById('breadcrumbs-header')
  if (breadcrumbContainer) {
    ReactDOM.render(<Breadcrumbs include="directory" language={language} />, breadcrumbContainer)
  }
}

function MainContent({ children }) {
  return (
    <Col>
      <main id="mainContent">{children}</main>
    </Col>
  )
}

function PageLayout({ children }) {
  const { language } = useStore()
  useEffect(() => renderBreadcrumbsIntoKthHeader(language))
  return (
    // Container in publicLayout.handlebars – begin
    <Row>
      <MainMenu />
      <MainContent>{children}</MainContent>
    </Row>
    // Container – end
  )
}

export default PageLayout
