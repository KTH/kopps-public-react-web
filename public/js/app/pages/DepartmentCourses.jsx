import React, { Fragment } from 'react'
import { Col, Row } from 'reactstrap'
import { Link, PageHeading, SortableTable } from '@kth/kth-reactstrap/dist/components/studinfo'

import Article from '../components/Article'
import FooterContent from '../components/FooterContent'

import { useStore } from '../mobx'
import i18n from '../../../../i18n'
import translate from '../../../../domain/translate'
import { courseLink } from '../util/links'

function codeCell(code) {
  const { language } = useStore()
  return {
    content: <Link href={courseLink(code, language)}>{code}</Link>,
    sortKey: code,
  }
}

function titleCell(code, title) {
  const { language } = useStore()
  return {
    content: <Link href={courseLink(code, language)}>{title}</Link>,
    sortKey: title,
  }
}

function compareCoursesBy(key) {
  return function compare(a, b) {
    if (a[key] < b[key]) {
      return -1
    }
    if (a[key] > b[key]) {
      return 1
    }
    return 0
  }
}

function sortAndParseCourses(courses) {
  courses.sort(compareCoursesBy('code'))
  const parsedCourses = courses.map(({ code, title, credits, creditUnitAbbr, level }) => [
    codeCell(code),
    titleCell(code, title),
    `${credits} ${creditUnitAbbr}`,
    level,
  ])
  return parsedCourses
}

function DepartmentsList() {
  const { language, departmentName, departmentCourses } = useStore()
  const t = translate(i18n, language)

  const headers = [
    t('department_course_code'),
    t('department_course_name'),
    t('department_course_credits'),
    t('department_course_educational_level'),
  ]
  const courses = sortAndParseCourses(departmentCourses)

  return (
    <>
      <Row>
        <Col>
          <PageHeading subHeading={departmentName}>{t('department_courses')}</PageHeading>
        </Col>
      </Row>
      <Row>
        <Col>
          <Article>
            <SortableTable headers={headers} data={courses} />
          </Article>
        </Col>
      </Row>
      <Row>
        <Col>
          <FooterContent />
        </Col>
      </Row>
    </>
  )
}

export default DepartmentsList
