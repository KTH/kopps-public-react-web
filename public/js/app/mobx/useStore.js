/* eslint no-use-before-define: ["error", "nofunc"] */

import createApplicationStore from '../stores/createApplicationStore'
import { getMobxManager } from './manager'

export default useStore

/**
 * Hook which can be used by any React component
 * within the related MobxStoreProvider.
 *
 * @param {string} [storeId]
 *
 *      Current state of application store
 */
function useStore(storeId = 'default') {
  const manager = getMobxManager(storeId)

  const isTestEnvironmentWithoutProviderComponent = manager == null
  if (isTestEnvironmentWithoutProviderComponent) {
    const dummyStoreForTesting = createApplicationStore()
    return dummyStoreForTesting
  }

  return manager.getContextHook()
}
