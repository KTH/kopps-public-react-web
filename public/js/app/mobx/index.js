/* eslint no-use-before-define: ["error", "nofunc"] */

import MobxStoreProvider from './MobxStoreProvider'
import useStore from './useStore'
import { compressStoreIntoJavascriptCode, uncompressStoreInPlaceFromDocument } from './compress'
// import { compressStoreIntoJavascriptCode, uncompressStoreInPlaceFromDocument } from './compressWithPako'

export { MobxStoreProvider, useStore, compressStoreIntoJavascriptCode, uncompressStoreInPlaceFromDocument }
