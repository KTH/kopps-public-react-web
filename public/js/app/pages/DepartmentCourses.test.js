/* eslint-disable react/prop-types */

import React from 'react'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { axe, toHaveNoViolations } from 'jest-axe'
import { StaticRouter } from 'react-router'
import { MobxStoreProvider } from '../mobx'

import DepartmentCourses from './DepartmentCourses'
import RouteWrapper from '../components/RouteWrapper'
import PageLayout from '../layout/PageLayout'

import createApplicationStore from '../stores/createApplicationStore'
import getMenuDepartmentData from '../config/departmentMenuData'

import commonSettings from '../config/mocks/mockCommonSettings'

expect.extend(toHaveNoViolations)

const applicationStore = createApplicationStore()

const testProxyPath = '/kopps-public'
const proxyPrefixPath = commonSettings.proxyPrefixPath
const testDepartmentName = {
  sv: 'ABE/Arkitektur, Fex övr utbildning',
  en: 'ABE/Architecture',
}
const testDepartmentCourses = {
  sv: [
    {
      code: 'AD12DA',
      title: '2DAutoCAD Fokus på ritningen som kommunikationsredskap',
      info:
        '<p>Kursens &#246;vergripande m&#229;l &#228;r att ge grundl&#228;ggande kunskap om Autocad som ritverktyg och om hur programmet fungerar som hj&#228;lpmedel f&#246;r att hantera olika slags arkitekturpojekt.</p><p>Efter genomf&#246;rd kurs skall studententen p&#229; egen hand kunna rita och utveckla arkitekturprojekt med hj&#228;lp av Autocad.</p>',
      credits: '6,0',
      creditUnitLabel: 'Högskolepoäng',
      creditUnitAbbr: 'hp',
      level: 'Grundnivå',
      state: 'CANCELLED',
      last_examination_semester: '20221',
    },
  ],
  en: [
    {
      code: 'AD12DA',
      title: '2DAutoCAD Focus on the Drawing as a Communicative Tool',
      info:
        '<p>The overall objective of the course is to give a basic knowledge of Autocad as a tool for drawing and of how the program can be used as a tool to handle different types of architecture projects.</p><p>On completing the course, the student shall be able to independently draw and develop architecture projects by using Autocad.</p>',
      credits: '6.0',
      creditUnitLabel: 'Credits',
      creditUnitAbbr: 'hp',
      level: 'First cycle',
      state: 'CANCELLED',
      last_examination_semester: '20221',
    },
  ],
}

const WrapperDepartmentCourses = ({ lang }) => {
  applicationStore.setLanguage(lang)
  applicationStore.setBrowserConfig({ proxyPrefixPath })

  const updatedApplicationStore = {
    ...applicationStore,
  }
  return (
    <StaticRouter>
      <RouteWrapper
        exact
        path="/student/kurser/org/ADD"
        createBreadcrumbs={() => ({ include: 'directory' })}
        component={DepartmentCourses}
        layout={PageLayout}
        applicationStore={updatedApplicationStore}
        createMenuData={store => ({ selectedId: 'courses', ...getMenuDepartmentData(store) })}
      />
    </StaticRouter>
  )
}

const DepartmentCoursesWithLayout = ({ lang }) => {
  applicationStore.setLanguage(lang)
  applicationStore.setBrowserConfig({ proxyPrefixPath })
  applicationStore.setDepartmentName(testDepartmentName[lang])
  applicationStore.setDepartmentCourses(testDepartmentCourses[lang])
  const departmentMenuData = getMenuDepartmentData(applicationStore)

  const updatedApplicationStore = {
    ...applicationStore,
  }
  return (
    <StaticRouter>
      <MobxStoreProvider initCallback={() => updatedApplicationStore}>
        <PageLayout breadcrumbs={{ include: 'directory' }} menuData={{ selectedId: 'courses', ...departmentMenuData }}>
          <DepartmentCourses />
        </PageLayout>
      </MobxStoreProvider>
    </StaticRouter>
  )
}

describe('Render component DepartmentCourses within RouterWrapper', () => {
  test('simple render', done => {
    render(<WrapperDepartmentCourses lang="en" />)
    done()
  })
})

describe('Render component DepartmentCourses within Layout', () => {
  test('verify that page is accessible', async () => {
    const { container } = render(<DepartmentCoursesWithLayout lang="en" />)
    expect(await axe(container)).toHaveNoViolations()
  })
  test('get page header in English', () => {
    render(<DepartmentCoursesWithLayout lang="en" />)
    const h1Header = screen.getByRole('heading', { level: 1 })
    expect(h1Header).toHaveTextContent('Courses')
  })
  test('get page header in Swedish', () => {
    render(<DepartmentCoursesWithLayout lang="sv" />)
    const h1Header = screen.getByRole('heading', { level: 1 })
    expect(h1Header).toHaveTextContent('Kurser')
  })
  test('match to snapshot in English', done => {
    const { asFragment } = render(<DepartmentCoursesWithLayout lang="en" />)
    expect(asFragment()).toMatchSnapshot()
    done()
  })
  test('match to snapshot in Swedish', done => {
    const { asFragment } = render(<DepartmentCoursesWithLayout lang="sv" />)
    expect(asFragment()).toMatchSnapshot()
    done()
  })
})

describe('Render component DepartmentCourses and check its menu, content and links', () => {
  test('check all links on the page in English', () => {
    render(<DepartmentCoursesWithLayout lang="en" />)
    const links = screen.getAllByRole('link')
    expect(links.length).toBe(5)
    // Menu links
    expect(links[0]).toHaveTextContent('Course and programme directory')
    expect(links[0].href).toStrictEqual('http://localhost/student/kurser/org?l=en')
    // Main content links
    expect(links[1]).toHaveTextContent('AD12DA')
    expect(links[1].href).toStrictEqual('http://localhost/student/kurser/kurs/AD12DA?l=en')
    expect(links[2]).toHaveTextContent('2DAutoCAD Focus on the Drawing as a Communicative Tool')
    expect(links[2].href).toStrictEqual('http://localhost/student/kurser/kurs/AD12DA?l=en')
    // Footer links
    expect(links[3]).toHaveTextContent('Central study counseling')
    expect(links[3].href).toStrictEqual('https://www.kth.se/studycounselling')
    expect(links[4]).toHaveTextContent('kopps@kth.se')
    expect(links[4].href).toStrictEqual('mailto:kopps@kth.se')
  })
  test('check all links on the page in Swedish', () => {
    render(<DepartmentCoursesWithLayout lang="sv" />)
    const links = screen.getAllByRole('link')
    expect(links.length).toBe(5)
    // Menu links
    expect(links[0]).toHaveTextContent('Kurs- och programkatalogen')
    expect(links[0].href).toStrictEqual('http://localhost/student/kurser/org')
    // Main content links
    expect(links[1]).toHaveTextContent('AD12DA')
    expect(links[1].href).toStrictEqual('http://localhost/student/kurser/kurs/AD12DA')
    expect(links[2]).toHaveTextContent('2DAutoCAD Fokus på ritningen som kommunikationsredskap')
    expect(links[2].href).toStrictEqual('http://localhost/student/kurser/kurs/AD12DA')
    // Footer links
    expect(links[3]).toHaveTextContent('Central studievägledning')
    expect(links[3].href).toStrictEqual('https://www.kth.se/studycounselling')
    expect(links[4]).toHaveTextContent('kopps@kth.se')
    expect(links[4].href).toStrictEqual('mailto:kopps@kth.se')
  })
})
