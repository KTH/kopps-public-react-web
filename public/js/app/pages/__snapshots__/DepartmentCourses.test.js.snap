// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render component DepartmentCourses within Layout match to snapshot in English 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <nav
      aria-labelledby="kth-local-navigation-title--mobile"
      class="kth-local-navigation--mobile"
    >
      <button
        class="kth-button menu"
        id="kth-local-navigation-title--mobile"
      >
        <span>
          ABE/Architecture
        </span>
      </button>
      <dialog
        class="kth-mobile-menu left"
      >
        <div
          class="kth-mobile-menu__navigation"
        >
          <button
            class="kth-icon-button close"
          >
            <span
              class="kth-visually-hidden"
            >
              Close
            </span>
          </button>
        </div>
        <div
          class="mobile-menu__content"
        >
          <a
            class="kth-button back"
            href="/student/kurser?l=en"
          >
            Course and programme directory
          </a>
          <span>
            ABE/Architecture
          </span>
          <ul>
            <li>
              <a
                aria-current="page"
              >
                Courses
              </a>
            </li>
          </ul>
        </div>
      </dialog>
    </nav>
    <nav
      aria-labelledby="local-navigation-title"
      class="kth-local-navigation col"
      id="mainMenu"
    >
      <a
        class="kth-button back"
        href="/student/kurser?l=en"
      >
        Course and programme directory
      </a>
      <h2
        id="local-navigation-title"
      >
        ABE/Architecture
      </h2>
      <ul>
        <li>
          <a
            aria-current="page"
          >
            Courses
          </a>
        </li>
      </ul>
    </nav>
    <main
      class="col"
      id="mainContent"
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <header
            role="presentation"
          >
            <h1
              aria-labelledby="page-heading page-sub-heading"
              id="page-heading"
            >
              Courses
            </h1>
            <p
              aria-hidden="true"
              class="page-sub-heading"
              id="page-sub-heading"
            >
              ABE/Architecture
            </p>
          </header>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <article
            class="article "
          >
            <table
              class="table table-responsive"
            >
              <thead>
                <tr>
                  <th
                    style="cursor: pointer; user-select: none;"
                  >
                    Course code
                    <span
                      aria-hidden="true"
                      style="opacity: 0; margin-left: .25em;"
                    >
                      ▼
                    </span>
                  </th>
                  <th
                    style="cursor: pointer; user-select: none;"
                  >
                    Course name
                    <span
                      aria-hidden="true"
                      style="opacity: 0; margin-left: .25em;"
                    >
                      ▼
                    </span>
                  </th>
                  <th
                    style="cursor: pointer; user-select: none;"
                  >
                    Scope
                    <span
                      aria-hidden="true"
                      style="opacity: 0; margin-left: .25em;"
                    >
                      ▼
                    </span>
                  </th>
                  <th
                    style="cursor: pointer; user-select: none;"
                  >
                    Educational level
                    <span
                      aria-hidden="true"
                      style="opacity: 0; margin-left: .25em;"
                    >
                      ▼
                    </span>
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>
                    <a
                      href="/student/kurser/kurs/AD12DA?l=en"
                    >
                      AD12DA
                    </a>
                  </td>
                  <td>
                    <a
                      href="/student/kurser/kurs/AD12DA?l=en"
                    >
                      2DAutoCAD Focus on the Drawing as a Communicative Tool
                    </a>
                  </td>
                  <td>
                    6.0 hp
                  </td>
                  <td>
                    First cycle
                  </td>
                </tr>
              </tbody>
            </table>
          </article>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <footer
            class="border-top mt-4 pt-1"
            id="articleFooter"
          >
            <address>
              Contact for questions regarding education at KTH: 
              <a
                href="https://www.kth.se/en/studies/master/general-study-counselling-1.621634"
              >
                Central study counseling
              </a>
              <br />
              Contact regarding technical matters on this page: 
              <a
                href="mailto:kopps@kth.se"
              >
                kopps@kth.se
              </a>
            </address>
          </footer>
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;

exports[`Render component DepartmentCourses within Layout match to snapshot in Swedish 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <nav
      aria-labelledby="kth-local-navigation-title--mobile"
      class="kth-local-navigation--mobile"
    >
      <button
        class="kth-button menu"
        id="kth-local-navigation-title--mobile"
      >
        <span>
          ABE/Arkitektur, Fex övr utbildning
        </span>
      </button>
      <dialog
        class="kth-mobile-menu left"
      >
        <div
          class="kth-mobile-menu__navigation"
        >
          <button
            class="kth-icon-button close"
          >
            <span
              class="kth-visually-hidden"
            >
              Close
            </span>
          </button>
        </div>
        <div
          class="mobile-menu__content"
        >
          <a
            class="kth-button back"
            href="/student/kurser"
          >
            Kurs- och programkatalogen
          </a>
          <span>
            ABE/Arkitektur, Fex övr utbildning
          </span>
          <ul>
            <li>
              <a
                aria-current="page"
              >
                Kurser
              </a>
            </li>
          </ul>
        </div>
      </dialog>
    </nav>
    <nav
      aria-labelledby="local-navigation-title"
      class="kth-local-navigation col"
      id="mainMenu"
    >
      <a
        class="kth-button back"
        href="/student/kurser"
      >
        Kurs- och programkatalogen
      </a>
      <h2
        id="local-navigation-title"
      >
        ABE/Arkitektur, Fex övr utbildning
      </h2>
      <ul>
        <li>
          <a
            aria-current="page"
          >
            Kurser
          </a>
        </li>
      </ul>
    </nav>
    <main
      class="col"
      id="mainContent"
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <header
            role="presentation"
          >
            <h1
              aria-labelledby="page-heading page-sub-heading"
              id="page-heading"
            >
              Kurser
            </h1>
            <p
              aria-hidden="true"
              class="page-sub-heading"
              id="page-sub-heading"
            >
              ABE/Arkitektur, Fex övr utbildning
            </p>
          </header>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <article
            class="article "
          >
            <table
              class="table table-responsive"
            >
              <thead>
                <tr>
                  <th
                    style="cursor: pointer; user-select: none;"
                  >
                    Kurskod
                    <span
                      aria-hidden="true"
                      style="opacity: 0; margin-left: .25em;"
                    >
                      ▼
                    </span>
                  </th>
                  <th
                    style="cursor: pointer; user-select: none;"
                  >
                    Kursnamn
                    <span
                      aria-hidden="true"
                      style="opacity: 0; margin-left: .25em;"
                    >
                      ▼
                    </span>
                  </th>
                  <th
                    style="cursor: pointer; user-select: none;"
                  >
                    Omfattning
                    <span
                      aria-hidden="true"
                      style="opacity: 0; margin-left: .25em;"
                    >
                      ▼
                    </span>
                  </th>
                  <th
                    style="cursor: pointer; user-select: none;"
                  >
                    Utbildningsnivå
                    <span
                      aria-hidden="true"
                      style="opacity: 0; margin-left: .25em;"
                    >
                      ▼
                    </span>
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>
                    <a
                      href="/student/kurser/kurs/AD12DA"
                    >
                      AD12DA
                    </a>
                  </td>
                  <td>
                    <a
                      href="/student/kurser/kurs/AD12DA"
                    >
                      2DAutoCAD Fokus på ritningen som kommunikationsredskap
                    </a>
                  </td>
                  <td>
                    6,0 hp
                  </td>
                  <td>
                    Grundnivå
                  </td>
                </tr>
              </tbody>
            </table>
          </article>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <footer
            class="border-top mt-4 pt-1"
            id="articleFooter"
          >
            <address>
              Kontakt för frågor om utbildning på KTH: 
              <a
                href="https://www.kth.se/studievagledning"
              >
                Central studievägledning
              </a>
              <br />
              Teknisk kontakt om denna sida: 
              <a
                href="mailto:kopps@kth.se"
              >
                kopps@kth.se
              </a>
            </address>
          </footer>
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
